src/global_data/4C_global_data_read.cpp

```diff
@@ -82,20 +82,23 @@ namespace
 
       auto valid_functions = functionmanager.valid_function_lines();
 
-      // TODO Move the function number inside the general function section to stop relying
-      // on explicit function numbers. The magic number 30 is equal to the highest number appearing
-      // in the test files.
-      for (unsigned i : std::views::iota(1, 30))
-      {
-        section_specs["FUNCT" + std::to_string(i)] = valid_functions;
-      }
+      // The FUNCT sections are special and do not fit into the usual pattern of sections and the
+      // capabilities of InputSpec. The special FUNCT<n> section is not supposed to be entered by
+      // users but we use this information inside the input file. Not pretty, but it works.
+      // TODO remove this hack by restructuring the input of functions.
+      section_specs["FUNCT<n>"] = valid_functions;
```

Comment:
@isteinbrecher @davidrudlstorfer Since I had an intermediate state that would have broken meshpy again, I opted for a new approach without the magic number.

@gilrrei I think we need a special case in schema for this. I am open to a different spelling of `FUNCT<n>`, if this makes this easier for you. 

Related to #375